core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/internal/checksums/factory/CrtBasedChecksumProvider.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.auth.aws.internal.checksums.factory;
+
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.zip.Checksum;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.utils.ClassLoaderHelper;
+import software.amazon.awssdk.utils.Lazy;
+import software.amazon.awssdk.utils.Logger;
+
+
+/**
+ * Class to load the Crt based checksum from aws-crt-java library if it is present in class path.
+ */
+@SdkInternalApi
+public final class CrtBasedChecksumProvider {
+
+    public static final Logger LOG = Logger.loggerFor(CrtBasedChecksumProvider.class);
+    private static final String CRT_CLASSPATH_FOR_CRC32C = "software.amazon.awssdk.crt.checksums.CRC32C";
+    private static final String CRT_CLASSPATH_FOR_CRC32 = "software.amazon.awssdk.crt.checksums.CRC32";
+    private static final Lazy<Optional<Class<?>>> CRT_CRC32_CLASS_LOADER =
+        new Lazy<>(() -> initializeCrtChecksumClass(CRT_CLASSPATH_FOR_CRC32));
+    private static final Lazy<Optional<Class<?>>> CRT_CRC32_C_CLASS_LOADER =
+        new Lazy<>(() -> initializeCrtChecksumClass(CRT_CLASSPATH_FOR_CRC32C));
+
+    private CrtBasedChecksumProvider() {
+    }
+
+    public static Checksum createCrc32() {
+        return createCrtBasedChecksum(CRT_CRC32_CLASS_LOADER);
+    }
+
+    public static Checksum createCrc32C() {
+        return createCrtBasedChecksum(CRT_CRC32_C_CLASS_LOADER);
+    }
+
+    private static Checksum createCrtBasedChecksum(Lazy<Optional<Class<?>>> lazyClassLoader) {
+        return lazyClassLoader.getValue().map(
+            checksumClass -> {
+                try {
+                    return (Checksum) checksumClass.getDeclaredConstructor().newInstance();
```

Comment:
I don't think we need to instantiate the class using reflection anymore now that crt is an optional dependency. Suggesting exposing something like below and using this method in Crc32CChecksum and Crc32Checksum

```
    public static boolean isCrtAvailable() {
        try {
            ClassLoaderHelper.loadClass(CRT_CLASSPATH_FOR_CRC32, false);
        } catch (ClassNotFoundException e) {
           return false;
        };
        
        return true;
    }
```

```
    public Crc32CChecksum() {
        if (isCrtAvailable()) {
            crc32c = new CRC32C();
        } else {
            crc32c = SdkCrc32C.create();
        }
    }
```